### DISCUSSION TITLE
Systemd Integration Restart Loop

### DISCUSSION THREAD

**Discussion by iamzoltan** [haugene/docker-transmission-openvpn#2070] [General]
hey, thanks for this great docker! It was exactly what I was looking for. I got the docker running and working and everything was A OK. The only trouble is that when I follow your instructions to try to turn transmission-openvpn into a systemd service, it constantly restarts and can not seem to stay running. Any ideas? Here is the .service file

[Unit]
Description=haugene/transmission-openvpn docker container
After=docker.service
Requires=docker.service

[Service]
User=myuser
TimeoutStartSec=0
ExecStartPre=-/usr/bin/docker kill transmission-openvpn
ExecStartPre=-/usr/bin/docker rm transmission-openvpn
ExecStartPre=/usr/bin/docker pull haugene/transmission-openvpn
ExecStart=/usr/bin/docker run \
        --name transmission-openvpn \
        --cap-add=NET_ADMIN -d \
        -v /home/myuser/data/:/data \
	-v /home/myuser/openvpn-credentials.txt:/config/openvpn-credentials.txt \
        -e OPENVPN_PROVIDER=EXPRESSVPN \
        -e OPENVPN_CONFIG=my_expressvpn_canada_-_montreal_-_2_udp,my_expressvpn_canada_-_montreal_udp,my_expressvpn_canada_-_toronto_-_2_udp,my_expressvpn_canada_-_toronto_udp,my_expressvpn_usa_-_new_jersey_-_1_udp,my_expressvpn_usa_-_new_jersey_-_2_udp,my_expressvpn_usa_-_new_jersey_-_3_udp,my_expressvpn_usa_-_new_york_-_2_udp,my_expressvpn_usa_-_new_york_udp,my_expressvpn_germany_-_frankfurt_-_1_udp,my_expressvpn_germany_-_frankfurt_-_2_udp,my_expressvpn_germany_-_frankfurt_-_3_udp,my_expressvpn_germany_-_nuremberg_udpn \
        -e OPENVPN_USERNAME=**None** \
        -e OPENVPN_PASSWORD=**None** \
  	-e TRANSMISSION_RPC_PASSWORD=******* \
  	-e TRANSMISSION_RPC_ENABLED=true \
  	-e TRANSMISSION_RPC_AUTHENTICATION_REQUIRED=true \
  	-e TRANSMISSION_RPC_USERNAME=****** \
        -e LOCAL_NETWORK=192.168.0.0/24 \
  	-e "TRANSMISSION_SCRAPE_PAUSED_TORRENTS_ENABLED=false" \
  	-e LOG_TO_STD=true \
        --log-driver json-file \
        --log-opt max-size=10m \
        -p 9091:9091 \
        --dns 8.8.8.8 \
        --dns 8.8.4.4 \
        haugene/transmission-openvpn
Restart=always
RestartSec=5

[Install]
WantedBy=multi-user.target

**Comment by iamzoltan** [comment]
haha, nevermind.... I think I found the issue. I removed the -d flag from this line:

--cap-add=NET_ADMIN -d

and it stopped restarting. :)

Any idea why that would happen? the man says its -d is detach mode, meaning its running in the background, but I'm not sure why that would cause the service to constantly restart. 

thanks for your time!

**Reply by pkishino** [reply]
Uhm, adding -d to the systemd doesn’t make sense though as it’s already in the background and no need for detach.. you can attach the log and we can see what it says though
